app/src/main/java/com/example/habitsmasher/NotificationFragment.java

```diff
@@ -0,0 +1,255 @@
+                                                                   _user.getId());
+
+        LinearLayoutManager layoutManager = new LinearLayoutManager(getContext(),
+                                                                        LinearLayoutManager.VERTICAL,
+                                                             false);
+        initializeRecyclerView(layoutManager, view);
+
+        return view;
+    }
+
+    @Override
+    protected void initializeRecyclerView(LinearLayoutManager layoutManager, View view) {
+        RecyclerView recyclerView = view.findViewById(R.id.notifications_recycler_view);
+        recyclerView.setItemAnimator(new DefaultItemAnimator());
+        recyclerView.setLayoutManager(layoutManager);
+        recyclerView.addItemDecoration(new DividerItemDecoration(getContext(), LinearLayoutManager.VERTICAL));
+        recyclerView.setAdapter(_notificationItemAdapter);
+
+        /* Implementation of swipe menu functionality came from this source:
+        Name: Velmurugan
+        Date: March 4, 2021
+        URL: https://howtodoandroid.com/android-recyclerview-swipe-menu
+         */
+        // create a touch listener which handles the click and swipe function of the RecyclerView
+        RecyclerTouchListener touchListener = new RecyclerTouchListener(getActivity(), recyclerView);
+        touchListener.setSwipeOptionViews(R.id.accept_request_button, R.id.deny_request_button)
+                .setSwipeable(R.id.requester_view, R.id.request_swipe_options, new RecyclerTouchListener.OnSwipeOptionsClickListener() {
+                            @Override
+                            public void onSwipeOptionClicked(int viewID, int position) {
+                                String followerID = _notificationItemAdapter._snapshots.get(position).getId();
+                                switch (viewID) {
+                                    case R.id.accept_request_button:
+                                        // accept the request
+                                        removeFollowRequestForUserInDatabase(_user.getId(),
+                                                followerID);
+                                        addNewFollowerForUserInDatabase(_user.getId(), followerID);
+                                        addUserToFollowingForUserInDatabase(followerID, _user.getId());
+                                        resetOptionsOfAdapter();
+                                        break;
+                                    case R.id.deny_request_button:
+                                        // deny the request
+                                        removeFollowRequestForUserInDatabase(_user.getId(),
+                                                _notificationItemAdapter._snapshots.get(position).getId());
+                                        resetOptionsOfAdapter();
+                                        break;
```

Comment:
When a request is accepted: Add the follower to the follower list of the target user, and add the followed user to the following list of the requesting user, and then delete the requesting user from the FollowRequests field
When a request is denied: Delete the requesting user from the FollowRequests field